nucypher/cli/painting/staking.py

```diff
@@ -267,3 +275,49 @@ def paint_min_rate(emitter, staker):
     ~ Previously set ....... {prettify_eth_amount(raw_minimum)}
     ~ Effective ............ {prettify_eth_amount(minimum)}"""
     emitter.echo(rate_payload)
+
+
+def paint_staking_rewards(stakeholder, blockchain, emitter, past_periods, staking_address, staking_agent):
+    if not past_periods:
+        reward_amount = stakeholder.staker.calculate_staking_reward()
+        emitter.echo(message=TOKEN_REWARD_CURRENT.format(reward_amount=round(reward_amount, TOKEN_DECIMAL_PLACE)))
+        return
+
+    economics = stakeholder.staker.economics
+    seconds_per_period = economics.seconds_per_period
+    current_period = staking_agent.get_current_period()
+    from_period = current_period - past_periods
+    latest_block = blockchain.client.block_number
+    from_block = estimate_block_number_for_period(period=from_period,
+                                                  seconds_per_period=seconds_per_period,
+                                                  latest_block=latest_block)
+
+    argument_filters = {'staker': staking_address}
+    event_type = staking_agent.contract.events['Minted']
+    entries = event_type.getLogs(fromBlock=from_block,
+                                 toBlock='latest',
+                                 argument_filters=argument_filters)
+
+    rows = []
+    rewards_total = 0
+    for event_record in entries:
+        token_reward = NU(event_record['args']['value'], 'NuNit').to_tokens()
+        period = event_record['args']['period']
+        date = datetime_at_period(period, seconds_per_period, start_of_period=True)
+        rows.append([
+            date.local_datetime().strftime("%b %d %Y"),
+            int(event_record['blockNumber']),
+            int(period),
+            round(token_reward, TOKEN_DECIMAL_PLACE),
+        ])
+        rewards_total += token_reward
+
+    if not rows:
+        emitter.echo(TOKEN_REWARD_NOT_FOUND)
+        return
+
+    periods_as_days = math.floor(economics.days_per_period * past_periods)
```

Comment:
I don't think you need `math.floor` anymore since the `economics.days_per_period` calculation already does a floor operation ('//')
```python
self.days_per_period = hours_per_period // 24
```

```suggestion
    periods_as_days = economics.days_per_period * past_periods
```